### PR TITLE
Feature/build requires profile

### PR DESCRIPTION
**Pull request by lasote**
Recommended to review in 3 parts:

1) `conans/client/cmake.py`, `conans/client/generators/cmake_common.py` and  `conans/client/conf/__init__.py` for configure CMake variables for cross build
2) `conans/test/model/profile_test.py` and `conans/model/profiles.py` Added new build_requires to profiles
3) Rest: install/get/propagate info from build_requires

### CODE REVIEW

File: conans/client/manager.py [hunk 1]
```diff
@@ -139,87 +141,83 @@ def download(self, reference, package_ids, remote=None):
...
+
+    def _get_graph_builder(self, loader, update, remote_proxy, build_dep_infos=None):
         local_search = None if update else self._search_manager
         resolver = RequireResolver(self._user_io.out, local_search, remote_proxy)
-        builder = DepsGraphBuilder(remote_proxy, self._user_io.out, loader, resolver)
-        deps_graph = builder.load(None, conanfile)
+        graph_builder = DepsGraphBuilder(remote_proxy, self._user_io.out, loader, resolver, build_dep_infos)
+        return graph_builder
+
+    def _fill_info(self, conanfile, is_txt):
```
Comment (memsharded) [inline conans/client/manager.py]: Could these lines go into ``_get_conanfile_object`` as well?

File: conans/client/deps_builder.py [hunk 2]
```diff
@@ -269,20 +280,17 @@ def get_graph_updates_info(self, deps_graph):
...
-        """ compute the dependencies graph for:
-        param conan_ref: ConanFileReference for installed conanfile or path to user one
-                         might be None for user conanfile.py or .txt
-        """
+    def load(self, conanfile):
+
         dep_graph = DepsGraph()
         # compute the conanfile entry point for this dependency graph
-        root_node = Node(conan_ref, conanfile)
+        root_node = Node(None, conanfile)
```
Comment (lasote) [inline conans/client/deps_builder.py]: I think always None was passed, so I remove it.

File: conans/client/generators/cmake_common.py [hunk 3]
```diff
@@ -54,6 +54,7 @@ def cmake_dependencies(dependencies, build_type=""):
 set(CONAN_EXE_LINKER_FLAGS{build_type} "{deps.exelinkflags} ${{CONAN_EXE_LINKER_FLAGS{build_type}}}")
 set(CONAN_C_FLAGS{build_type} "{deps.cflags} ${{CONAN_C_FLAGS{build_type}}}")
 set(CONAN_CMAKE_MODULE_PATH{build_type} {deps.build_paths} ${{CONAN_CMAKE_MODULE_PATH{build_type}}})
+set(CONAN_CMAKE_FIND_ROOT_PATH{build_type} ${{CONAN_CMAKE_FIND_ROOT_PATH{build_type}}})
```
Comment (lasote) [inline conans/client/generators/cmake_common.py]: LoL, I tried to assign the lib folders in a crazy test, but it had no sense.

File: conans/client/manager.py (hunk 1 above)
Comment (lasote) [inline conans/client/manager.py]: It's used too in the info method. But I can move the invocation.

File: conans/client/loader.py [hunk 4]
```diff
@@ -200,14 +200,18 @@ def parse_conan_txt(self, contents, path, output):
...
-        conanfile.requires.add(str(reference))  # Convert to string necessary
-        # conanfile.options.values = options
-        self._user_options.scope_options(reference.name)
+        # Assign environment
+        conanfile._env_values.update(self._env_values)
+
+        for ref in references:
+            conanfile.requires.add(str(ref))  # Convert to string necessary
+            # conanfile.options.values = options
+            self._user_options.scope_options(ref.name)
```
Comment (lasote) [inline conans/client/loader.py]: No, it's a bad copy paste. I only wanted to apply the options declared in the profile, but I think it's no necessary to do anything there, right? Just pass to install?

Comment (lasote) [inline conans/client/loader.py]: Actually, I don't understand what's doing but it seems that those option lines are necessary if don't some test fail.

File: conans/client/deps_builder.py (hunk 2 above)
Comment (memsharded) [inline conans/client/deps_builder.py]: Yes, that is right, I have reviewed its usage, and it seems all collaborators using the consumer name, like "PROJECT" are receiving it as parameter.

File: conans/client/loader.py (hunk 4 above)
Comment (lasote) [inline conans/client/loader.py]: I've checked this. It was introduced (in the past) to allow options without package namespace in conan install commands with reference:

`conan install zlib/1.2.8@lasote/stable shared=True`

File: conans/client/generators/cmake_common.py [hunk 5]
```diff
@@ -381,6 +381,13 @@ def generate_targets_section(dependencies):
 
     # Make find_package() to work
     set(CMAKE_PREFIX_PATH ${CONAN_CMAKE_MODULE_PATH} ${CMAKE_PREFIX_PATH})
+
+    # Set the find root path (cross build)
+    set(CMAKE_FIND_ROOT_PATH ${CONAN_CMAKE_FIND_ROOT_PATH} ${CMAKE_FIND_ROOT_PATH})
+    set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM ${CONAN_CMAKE_FIND_ROOT_PATH_MODE_PROGRAM})
```
Comment (memsharded) [inline conans/client/generators/cmake_common.py]: This will overwrite existing user values, caution.

File: conans/client/manager.py (hunk 1 above)
Comment (memsharded) [inline conans/client/manager.py]: Ok, lets move this forward, we can figure out this later.

**Pull request merged**